fix memory leak in ares_uri (#1012)
Dynamic memory, referenced by 'outpath', is allocated at ares_uri.c:527
by calling function 'ares_buf_create' and lost at ares_uri.c:536.

Signed-off-by: Felix The Cat (@F3lixTheCat)

--

Upstream-Status: Backport [https://github.com/c-ares/c-ares/commit/ee2a1c3eff3c8164b09123005f4b49c571788b59]

--- a/src/lib/util/ares_uri.c	2025-08-10 15:59:17.501710307 -0400
+++ b/src/lib/util/ares_uri.c	2025-08-10 16:00:35.867494819 -0400
@@ -533,7 +533,7 @@
   status = ares_buf_split_str_array(inpath, (const unsigned char *)"/", 1,
                                     ARES_BUF_SPLIT_TRIM, 0, &arr);
   if (status != ARES_SUCCESS) {
-    return NULL;
+    goto done;
   }
 
   for (i = 0; i < (ares_ssize_t)ares_array_len(arr); i++) {
